### ISSUE TITLE
Plain logging contains rich color codes

### ISSUE THREAD

**Issue by sbrugman** [kedro-org/kedro#3626]
## Description
Following the instructions [here](https://docs.kedro.org/en/stable/logging/index.html#how-to-use-plain-console-logging), the logging will still outputs color codes.

## Expected Result

```txt
Loading data from datasetname (MyDataset).
```

## Actual Result
```txt
Loading data from [dark_orange]datasetname[/dark_orange] (MyDataset).
```

## Your Environment

* Kedro version used (0.19.2):

**Comment by sbrugman** [comment]
The markup should either be added conditional on that the rich handler is used, or stripped afterwards.

Instead of including the markup directly in the log, it could be parametrized:
```python
def fmt(value: str, color: str):
    return f"[{color}]{value}[/{color}]"

...

# Configurable, inferred from the logging handler, or conditional on `rich` installed.
markup = False

  self._logger.info(
      "Loading data from %s (%s)...",
      markup(name, color="dark_orange") if markup else name,
      type(dataset).__name__,
      extra={"markup": markup},
  )
```

instead of 
```python
        self._logger.info(
            "Loading data from [dark_orange]%s[/dark_orange] (%s)...",
            name,
            type(dataset).__name__,
            extra={"markup": True},
        )
```
https://github.com/kedro-org/kedro/blob/main/kedro/io/data_catalog.py#L487

For my use case it would be enough to disable the markup if `rich` is not installed, but no strong preference.

Alternatively, you could include a NoRichHandler that strips out the formatting. This would be unintuitive (why need to do additional work for requiring _fewer_ features), but it works.
 
Rich markup reference: https://rich.readthedocs.io/en/stable/markup.html

**Comment by noklam** [comment]
I think a simple condition is enough here, #3657 will not be merged soon feel free to open a PR @sbrugman.

**Comment by astrojuanlu** [comment]
I assume this was fixed in #3682.
